### PR TITLE
Cleanup technical debt reported by Sonarcloud

### PR DESCRIPTION
**Pull request by MrTimscampi**
* Removes the literal used for card shapes and creates a CardShapes enum
* Use proper casing for all enums
* Rename plugins to fit the semi-used standard already in place
* Fix an unneeded "undefined" on an option property in a type
* Reduce the cognitive complexity of one function in the codec detection (There's more, but since that's already touched in other PRs, I'm hesitant to do more there for now)

I also went over a bunch of Sonarcloud errors on their end to mark them as false positives or won't fix (It doesn't understand our absolute local imports, for example).

### CODE REVIEW

**Comment by dkanada** [comment]
Lowercase? I usually expect to see titlecasing or uppercasing for enums or constants like this one.

**Comment by MrTimscampi** [comment]
> Lowercase? I usually expect to see titlecasing or uppercasing for enums or constants like this one.

Hm, true, I didn't really pay attention :smile:

File: client/components/Skeletons/SkeletonCard.vue [hunk 1]
```diff
@@ -25,9 +26,9 @@ export default Vue.extend({
     },
     cardShape: {
       type: String,
-      default: (): string => 'portrait-card',
+      default: (): string => CardShapes.Portrait,
       validator: (value): boolean =>
```
[the remark this answers is not in the document]
Comment (MrTimscampi) [reply client/components/Skeletons/SkeletonCard.vue:30]: The value is inferred from the definition of Vue itself. `validator` is a standard function, with its own signature.

File: client/components/Skeletons/SkeletonHomeSection.vue [hunk 2]
```diff
@@ -15,26 +15,27 @@
...
+import { CardShapes } from '~/utils/items';
 
 export default Vue.extend({
   props: {
     cardShape: {
       type: String,
-      validator: (val): boolean =>
-        ['thumb-card', 'portrait-card', 'square-card'].includes(val),
-      default: 'thumb-card'
+      validator: (value): boolean =>
```
[the remark this answers is not in the document]
Comment (MrTimscampi) [reply client/components/Skeletons/SkeletonHomeSection.vue:24]: Same as above.

**Review by ThibaultNocchi — APPROVED** [review]

**Review by ferferga — APPROVED** [review]
LGTM

**Comment by MrTimscampi** [comment]
@jellyfin-bot rebase

**Pull request merged**